### ISSUE TITLE
rename?

### ISSUE THREAD

**Issue by bmcmahen** [bmcmahen/react-wysiwyg#13]
this library has a wider usage than just wysiwyg. it's more accurately described as a content-editable wrapper that makes it easy to control and manipulate user-input.

**Comment by dtuite** [comment]
Definitely a good idea. And worth doing before the library gets too popular.

When I think of a "wysiwyg editor" I think of a huge plugin like [CKEditor](http://ckeditor.com/). That's not what this is.

**Comment by rjksn** [comment]
Since I'm looking for a WYSIWYG, I'm really confused that this is even listed as such.

**Comment by bmcmahen** [comment]
Yeah, I'll change this. Just too lazy right now... Thoughts on names? react-editable-input?

**Comment by dtuite** [comment]
react-contenteditable makes sense to me.

**Comment by bmcmahen** [comment]
agreed, unfortunately it's already taken on npm.

**Comment by rjksn** [comment]
So this is a contentEditable enabled element editor.

react-contentEditor? Or, go old school, w/ react-ContentEditr ;)

**Comment by CoreyTrombley** [comment]
So i just want to point out that [this](https://github.com/vasanthk/react-inline-edit) was ripped off from you. even the `README.md`. :anger: :goberserk: So I would suggest seing if you can get that name, or bite that name in some way. That is was showed up first when I googled _react inline edit_.
Cheers! :beers:

**Comment by vasanthk** [comment]
@CoreyTrombley Sry if it came across as something unethical. 
I created that repo just for my own learning whenI was gettign started with React and modified it a bit to suit my project's needs. I have contributed to this project as well earlier when I did that. Mentioned it in the ReadMe as well.

**Comment by CoreyTrombley** [comment]
@vasanthk I sorry to judge so harshly, that was not really my intent, but it just rubbed me the wrong way when I went to edit your readme to make a little more sense, and googled what contentEditable was and found his blog post. It is my fault for jumping to conclusions and not reading thoroughly though, so I apologise. :sweat_smile:

We should just come together as it seems we all have the same idea/intent with this type of component.

**Comment by bmcmahen** [comment]
Yeah, the original intent of this module was really simple: Allow the user to perform inline-editing in a matter that emulated an input field or textarea, but retained the original styles of the non-edited state. `react-inline-edit` is actually a much better name, heh. But recently it's been rewritten a bit to make it more flexible -- i.e., to allow tweet-box type functionality, as shown in the demo.

`react-contenteditor` isn't bad. Hmm...

**Comment by vasanthk** [comment]
@bmcmahen Feel free to use the react-inline-edit name if you need to :)

**Comment by vasanthk** [comment]
On a different note, I played around with `contenteditable` a bit more and ran into several cross browser issues, especially when I tried to use it in an inline edit form and tried to use keyboard shortcuts to tab between fields etc. 
Behaves weirdly in non-Chrome browsers. Ended up using an autosizeable textarea with styles to make it appear as an inline edit.

**Comment by bmcmahen** [comment]
Thanks @vasanthk. I agree about using textareas and form inputs when possible. `contenteditable` is a disaster for cross-browser compatibility, not to mention things like accessibility. I think it's worth sacrificing the potential slick feel of inline-editing functionality for the more reliable input and textarea fields. 

The problem with traditional form inputs is for doing anything more complicated, like the tweet-box example that I provide. If you want to do something like that, you're unfortunately stuck with `contenteditable`

**Comment by vasanthk** [comment]
@bmcmahen With you on that!
@CoreyTrombley No worries :)
